feat: expose request_class to other layers (#3125)
litestar-org/litestar: docs/usage/requests.rst

.. literalinclude:: /examples/request_data/msgpack_request.py
   :caption: msgpack_request.py
   :language: python

> Custom Request
> ----------------
> 

Reviewer: ```suggestion
Custom Request
--------------

.. versionadded:: 2.7.0

```

> Litestar supports custom ``request_class`` which can enrich default :class:`Request`.
> The example below illustrates how to implement custom request class for the whole application.
> 

Reviewer: ```suggestion
Litestar supports custom ``request_class``es which can be used to further configure the default :class:`Request`.
The example below illustrates how to implement custom request class for the whole application.

```

.. literalinclude:: /examples/request_data/custom_request.py
    :language: python

.. admonition:: Layered architecture

   Request classes are part of Litestar's layered architecture, which means you can
   set a request class on every layer of the application. If you have set a request
   class on multiple layers, the layer closest to the route handler will take precedence.

>    You can read more about this here: :ref:`usage/applications:layered architecture`

Reviewer: ```suggestion
   You can read more about this in the: :ref:`usage/applications:layered architecture` section
```

.. literalinclude:: /examples/request_data/msgpack_request.py
   :caption: msgpack_request.py
   :language: python

Custom Request
--------------

.. versionadded:: 2.7.0

> Litestar supports custom ``request_class``es which can be used to further configure the default :class:`Request`.

Reviewer: oops broke thing.. sorry

The example below illustrates how to implement custom request class for the whole application.

> .. literalinclude:: /examples/request_data/custom_request.py
>     :language: python

Reviewer: ```suggestion
.. dropdown:: Example of a custom request at the application level

    .. literalinclude:: /examples/request_data/custom_request.py
        :language: python
```

Reviewer: I cant test this at the moment but hiding this in a dropdown might be good to save some space if not needing this info

.. admonition:: Layered architecture

   Request classes are part of Litestar's layered architecture, which means you can
   set a request class on every layer of the application. If you have set a request
   class on multiple layers, the layer closest to the route handler will take precedence.

>    You can read more about this in the: :ref:`usage/applications:layered architecture` section

Reviewer: ```suggestion
   You can read more about this in the :ref:`usage/applications:layered architecture` section
```

.. literalinclude:: /examples/request_data/msgpack_request.py
   :caption: msgpack_request.py
   :language: python

Custom Request
--------------

> .. versionadded:: 2.7.0

Reviewer: I agree on the consistency but I was also thinking that if we start it, then it would never be done or require a lot of digging into when things were added to go back and add them.

Im up in the air on this
